tests/ui/lint/removed-lints/README.md

```diff
@@ -0,0 +1,4 @@
+# Removed lints
+
+This directory contains tests to confirm that lints that have been
+removed do not cause errors and produce the appropriate warnings.
```

Comment:
How would that ever prevent a problem?

As-in, if you forgot to remove the lint properly surely you'd also forget to add a test? And it seems very unlikely that any compiler change would break removed lints...

Reply:
Yes, I had the same thoughts, but... you know, these removed lints are actually part of our language.  If there were a "Rust-compatibility test suite", these should be in there.  So, anyway, that pushed me slightly over the line to adding this.

Also, when reviewing quickly, like we do on the lang side during calls, we tend to focus on the tests, so it's helpful to see everything we believe should be true actually double-checked there.

Reply:
If we are to proceed with it, could you open an issue to back-fill removed-lints tests for all previously removed lints?

Reply:
Sure, I'll open an issue after we merge this.